scripts/useReactVersion.mjs

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-console */
+/**
+ * Given the dist tag fetch the corresponding
+ * version and make sure this version is used throughout the repository.
+ *
+ * If you work on this file:
+ * WARNING: This script can only use built-in modules since it has to run before
+ * `pnpm install`
+ */
+import childProcess from 'child_process';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { promisify } from 'util';
+import { getWorkspaceRoot } from './utils.mjs';
+
+// TODO: reuse the `useReactVersion.mjs` from the monorepo
+
+const exec = promisify(childProcess.exec);
+
+// packages published from the react monorepo using the same version
+const reactPackageNames = ['react', 'react-dom', 'react-is', 'react-test-renderer', 'scheduler'];
+const devDependenciesPackageNames = ['@mnajdova/enzyme-adapter-react-18', '@testing-library/react'];
```

Comment:
Feel free to adapt similar changes as in: https://github.com/mui/material-ui/pull/42706/files#diff-c7b5ce1d178ce80c5ed1c5592d04aa583904c77e4804b504442cec82e799928e because https://github.com/mui/mui-x/pull/13662 has been merged.